### PR TITLE
Disable NTP time compensation

### PR DESCRIPTION
**Pull request by ianshade**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature change

* **What is the current behavior?** (You can also link to an open issue here)
Sofie Core periodically calculates how "wrong" its system clock is in comparison to the NTP server, and in `getCurrentTime` it compensates the values used in internal calculations that normally would just use `Date.now()`.
This is believed to be an incorrect approach for multiple reasons.

* **What is the new behavior (if this is a feature change)?**
Sofie still establishes the diff between system time and the NTP server time, but does not use the diff in any other calculations. It serves information purpose only. Logging and system status is updated accordingly. 
From now on we're assuming that the system clock of the server is NTP-synced and accurate enough for the calculations, meaning `getCurrentTime` on the server side can return `Date.now()`. (but it will be detected if for some reason it isn't)
Syncing the time between server and the client, and between server and peripheral devices stays as it was.

* **Other information**:

**Status**

- [x] Code documentation for the relevant parts in the code have been added/updated by the PR author
- [x] The functionality has been tested by the PR author
- [ ] The functionality has been tested by NRK

### CODE REVIEW

**Review by nytamin — CHANGES_REQUESTED** [review]
I'm definitely open for a discussion regarding the existence of the Sofie-NTP-sync feature, but right now I do not think we should remove it altogether, but rather make it an opt-in setting.

The current functionality is:
* NTP sync is always on
* NTP-servers are defined by `env.NTP_SERVERS`

I propose that we instead:

* NTP-sync is off by default
* Add the following properties to the `Settings`:
```
{
  /** If enabled, Core will automatically sync its time to the NTP servers specified. */
  enableNTPSync: false | [string] // Default: env.NTP_SERVERS or false
  /** Maximum allowed deviation (sync quality) from a time sync */
  maxAllowedTimeDev: number // Default: 50 
}
```

File: meteor/server/api/systemTime/systemTime.ts [hunk 1]
```diff
@@ -146,18 +146,17 @@ function updateServerTime(retries: number = 0) {
...
+				} ms)`
+				logger.info(`System time: ${message}`)
 
 				systemTime.hasBeenSet = true
 				systemTime.diff = result.mean
 				systemTime.stdDev = result.stdDev
 				setSystemStatus('systemTime', {
-					statusCode: StatusCode.GOOD,
-					messages: [`NTP-time accuracy (standard deviation): ${Math.floor(result.stdDev * 10) / 10} ms`],
+					statusCode: systemTime.diff < 200 ? StatusCode.GOOD : StatusCode.WARNING_MAJOR,
```
[the remark this answers is not in the document]
Comment (ianshade) [reply meteor/server/api/systemTime/systemTime.ts:158]: It is a measure of the quality / presence of the NTP client in the operating system, nothing else.

**Comment by ianshade** [comment]
If we want to keep this functionality, we either should fix it, or make it very clear that the only case when this might work as intended, is when the OS has no time synchronization enabled - no NTP client, no periodic time.windows.com checks, nothing, because otherwise it's just fighting against what the system does

**Comment by Julusian** [comment]
> I think, that if we want to keep this functionality, we either should fix it

@ianshade I'm curious how you think this can be fixed? As far as I can see the big problem with the NTP logic here is that it fights the system NTP client, but aside from that it works.

@nytamin What do we benefit by keeping the syncing? It will be very poorly tested as it will be opt-in and neither nrk or tv2 will have it enabled in production. I doubt any developers will enable it locally either. So who will this feature be for?

**Comment by ianshade** [comment]
We should calculate the diff more often. We could add some low-level code comparing the realtime and monotonic clocks to detect whenever the time is adjusted on the system level, and recalculate the diff shortly after.
The playout-gateway and clients should be immediately made aware of large diffs because it may affect them and cause bugs.

**Comment by nytamin** [comment]
Note: After some discussion between me, @ianshade and @Julusian we agreed to remove the NTP sync from Core, but also to add a separate functionality to check the system-time-diff against an external NTP server (to raise warnings only).

**Pull request merged**